src/Settings.js

```diff
@@ -56,23 +56,42 @@ export class Settings {
             const autofillSettings = validate(runtimeConfig?.userPreferences?.features?.autofill?.settings, autofillSettingsSchema)
             return autofillSettings.featureToggles
         } catch (e) {
-            // these are the fallbacks for when a platform hasn't implemented the calls above. (like on android)
+            // these are the fallbacks for when a platform hasn't implemented the calls above.
             if (this.globalConfig.isDDGTestMode) {
                 console.log('isDDGTestMode: getFeatureToggles: ❌', e)
             }
             return Settings.defaults.featureToggles
         }
     }
 
+    /**
+     * Get the availableInputTypes coming from the global configs
+     * @returns {Promise<AvailableInputTypes>}
+     */
+    async getInitialAvailableInputTypes () {
```

Comment:
This is similar to what we're already doing with feature flags. We're reading this at startup. The old `getAvailableInputTypes` is now meant to be called only when needed and by passing a specific type.